include/seqan3/alignment/aligned_sequence/aligned_sequence_concept.hpp

```diff
@@ -272,4 +285,93 @@ inline typename seq_type::iterator erase_gap(seq_type & seq, typename seq_type::
 }
 //!\}
 
+namespace detail
+{
+
+/*!\brief               Create the formatted alignment output and add it to the provided debug_stream.
+ * \ingroup             aligned_sequence
+ * \tparam alignment_t  The type of the alignment, must satisfy tuple_like_concept.
+ * \tparam idx          An index sequence.
+ * \param[in] stream    The output stream that receives the formatted alignment.
+ * \param[in] align     The alignment that shall be streamed.
+ */
+template<tuple_like_concept alignment_t, size_t ...idx>
+void stream_alignment(debug_stream_type & stream, alignment_t const & align, std::index_sequence<idx...> const & /**/)
+{
+    size_t const alignment_size = std::get<0>(align).size();
+    char const * indent = "        ";
+
+    // split alignment into blocks of length 50 and loop over parts
+    for (size_t begin_pos = 0; begin_pos < alignment_size; begin_pos += 50)
+    {
+        size_t const end_pos = std::min(begin_pos + 50, alignment_size);
+
+        // write header line
+        if (begin_pos != 0)
+            stream << '\n';
+
+        stream << std::setw(7) << begin_pos << ' ';
+        for (size_t rel_pos = 1; rel_pos <= (end_pos - 1) % 50 + 1; ++rel_pos)
+        {
+            if (rel_pos % 10 == 0)
+                stream << ':';
+            else if (rel_pos % 5 == 0)
+                stream << '.';
+            else
+                stream << ' ';
+        }
+
+        // write first sequence
+        stream << '\n' << indent;
+        ranges::for_each(std::get<0>(align) | ranges::view::slice(begin_pos, end_pos) | view::to_char,
+                         [&stream] (char ch) { stream << ch; });
+
+        auto stream_f = [&] (auto const & previous_seq, auto const & aligned_seq)
+        {
+            // write alignment bars
+            stream << '\n' << indent;
+            ranges::for_each(ranges::zip_view(previous_seq, aligned_seq) | ranges::view::slice(begin_pos, end_pos),
+                             [&stream] (auto ch) { stream << (std::get<0>(ch) == std::get<1>(ch) ? '|' : ' '); });
```

Comment:
the zip stream returns a rvalue. In order to safe a copy you should capture it per `&&`